test/integration/scheduler/scheduler_test.go

```diff
@@ -17,80 +17,119 @@ limitations under the License.
 package scheduler
 
 import (
-	"time"
-
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega"
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 
+	kueue "sigs.k8s.io/kueue/api/v1alpha1"
 	"sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/integration/framework"
 )
 
 // +kubebuilder:docs-gen:collapse=Imports
 
 var _ = ginkgo.Describe("Scheduler", func() {
 	const (
-		namespace   = "default"
-		instanceKey = "cloud.provider.com/instance"
-
-		timeout            = time.Second * 10
-		consistentDuration = time.Second * 3
-		interval           = time.Millisecond * 250
-		onDemandFlavor     = "on-demand"
-		spotFlavor         = "spot"
+		instanceKey    = "cloud.provider.com/instance"
+		onDemandFlavor = "on-demand"
+		spotFlavor     = "spot"
 	)
 
-	ginkgo.It("Should schedule jobs as they fit in their capacities", func() {
-		ginkgo.By("creating capacities and queues")
-		prodCapacity := testing.MakeCapacity("prod-capacity").
+	var (
+		ns           *corev1.Namespace
+		prodCapacity *kueue.Capacity
+		devCapacity  *kueue.Capacity
+		prodQueue    *kueue.Queue
+		devQueue     *kueue.Queue
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		if ns != nil {
```

Comment:
let's have this nil check as part of the framework

Reply:
done.